### DISCUSSION TITLE
Azure DNS zone: Certificate issuing fails -"The delegated name server is not correct."

### DISCUSSION THREAD

**Discussion by PEllitsgaard** [polymind-inc/acmebot#629] [Q&A] [answered]
Seems to suddenly having problems with getting certificates issued due to Azure DNS Zone returns empty string. The error is this:

**Orchestrator function 'IssueCertificate' failed: The activity function 'Dns01Precondition' failed: "The delegated name server is not correct. DNS zone = pubxyzz.com, Expected = ns1-04.azure-dns.com,ns2-04.azure-dns.net,ns3-04.azure-dns.org,ns4-04.azure-dns.info, Actual = ". See the function execution logs for additional details.**

I have already successfully got two certificates issued from DNS zones hosted in Azure. A couple of DNS Zones now fails with error similar to above.

All DNS zones that both are successful and fails has been assigned same role (as DNS zone contributor)

I have already tried to start, stop and restart the certbot function without any change in error message.

Any ideas?

**Comment by shibayan** [comment, marked as answer]
The error message suggests that the name servers in the DNS zone cannot be drawn from other DNS servers or are not configured correctly.

Execute the following command to check if the NS record values are returned correctly.

```
nslookup -type=NS <your domain name>
```

If set up correctly, the following results are returned.

![image](https://github.com/shibayan/keyvault-acmebot/assets/1356444/c8eee8ac-b0df-41cb-8f0a-f42f16df25e5)
